api/v1/Store.php

```diff
@@ -404,87 +429,106 @@ private function LoadJSON(string $jsonString, string $filename, string $dataStor
-                        }
-                        $this->insertMultiPolygon(
-                            $multi,
-                            $layerId,
-                            $encodedFeatureData,
-                            $countryId,
-                            $type,
-                            $mspId,
-                            $filename
-                        );
-                    }
-                } elseif (strcasecmp($geometryData["type"], "Point") == 0) {
+                continue;
+            }
+            if (strcasecmp($geometryData["type"], "MultiPoint") == 0) {
+                $this->InsertGeometry(
+                    $layerId,
+                    json_encode($geometryData["coordinates"]),
+                    $encodedFeatureData,
+                    $countryId,
+                    $type,
+                    $mspId,
+                    0,
+                    $filename
+                );
+                continue;
+            }
+            if (strcasecmp($geometryData["type"], "MultiLineString") == 0) {
+                foreach ($geometryData["coordinates"] as $line) {
                     $this->InsertGeometry(
                         $layerId,
-                        json_encode(array($geometryData["coordinates"])),
+                        json_encode($line),
                         $encodedFeatureData,
                         $countryId,
                         $type,
                         $mspId,
                         0,
                         $filename
                     );
-                } elseif (strcasecmp($geometryData["type"], "MultiLineString") == 0) {
-                    foreach ($geometryData["coordinates"] as $line) {
-                        $this->InsertGeometry(
-                            $layerId,
-                            json_encode($line),
-                            $encodedFeatureData,
-                            $countryId,
-                            $type,
-                            $mspId,
-                            0,
-                            $filename
-                        );
-                    }
-                } else {
-                    throw new Exception(
-                        "Encountered unknown feature type ".$geometryData["type"]." in layer ".$filename
-                    );
                 }
+                continue;
             }
+            throw new Exception(
+                "Encountered unknown feature type ".$geometryData["type"]." in layer ".$filename
+            );
         }
     }
 
+    /**
+     * @throws Exception
+     */
+    private function getLayerId(
+        string $layerName,
+        string $layerGeoType,
+        string $layerGroup
+    ): int {
+        $checkExists = Database::GetInstance()->query(
+            "SELECT layer_id FROM layer WHERE layer_name = ?",
+            array($layerName)
+        );
+        return $checkExists[0]["layer_id"] ?? Database::GetInstance()->query(
```

Comment:
This is fine, since if "layer_id" exists it should always be an int larger than 0 right ? If not if fall-back to the query.

Just to let you know. Before we used "empty", so you could also use ?: instead of ??

?? only evaluate to false with the value is null. Keep that in mind. So if the value would be 0, or false, or an empty array, all will evaluate to true.

Both "empty" and ?: will evaluate to false if the value is false after bool conversion, see "Converting to boolean" in
https://www.php.net/manual/en/language.types.boolean.php#language.types.boolean.casting.